sphinx/config.py

```diff
@@ -258,6 +258,7 @@ class Config:
         'math_number_all': _Opt(False, 'env', ()),
         'math_eqref_format': _Opt(None, 'env', frozenset((str,))),
         'math_numfig': _Opt(True, 'env', ()),
+        'math_numsep': _Opt(None, 'env', frozenset((str,))),
```

Comment:
Probably makes sense to unconditionally use ``math_numsep``, and have a sensible default. If a user sets ``math_numsep`` to ``None`` or ``''``, perhaps issue a warning/error, or silently use ``'.'``.
```suggestion
        'math_numsep': _Opt('.', 'env', frozenset((str,))),
```

Reply:
I've implemented the requested default separator, added to the CHANGES, and updated the documentation.